### PR TITLE
feat: implement editable dataframe to manage dataframe editor component

### PR DESCRIPTION
**Pull request by FabienArcellier**
implement https://github.com/FabienArcellier/writer-framework/issues/65

Peek 2024-07-06 08-36

### CODE REVIEW

**Comment by FabienArcellier** [comment]
You have to implement and document `state['df'].record(12) # get a record as dict`

File: src/writer/core.py [hunk 1]
```diff
@@ -1506,6 +1561,333 @@ def __set__(self, instance, value):
...
+    PandasRecordProcessor processes records from a pandas dataframe saved into an EditableDataframe
+
+    >>> df = pandas.DataFrame({"a": [1, 2], "b": [3, 4]})
+    >>> edf = EditableDataframe(df)
+    >>> edf.record_add({"a": 1, "b": 2})
+    """
+
+    @staticmethod
+    def match(df: Any) -> bool:
+        import pandas
```
Comment (FabienArcellier) [inline src/writer/core.py:1636]: manage missing pandas

File: src/writer/core.py [hunk 2]
```diff
@@ -1506,6 +1561,333 @@ def __set__(self, instance, value):
...
+    PolarRecordProcessor processes records from a polar dataframe saved into an EditableDataframe
+
+    >>> df = polars.DataFrame({"a": [1, 2], "b": [3, 4]})
+    >>> edf = EditableDataframe(df)
+    >>> edf.record_add({"record": {"a": 1, "b": 2}})
+    """
+
+    @staticmethod
+    def match(df: Any) -> bool:
+        import polars
```
Comment (FabienArcellier) [inline src/writer/core.py:1706]: manage missing polars

File: docs/framework/dataframe.mdx [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+	if payload.action == 'remove':
+	    state['mydf'].record_remove(payload)
+	if payload.action == 'important':
+		state['mydf'].record(payload.id).update('flag', True) # update the column flag of the dataframe to true, trigger une mutation record_update
+	if payload.action == 'open':
+		state['record'] = state['df'].record(payload.id)
+
+df = pandas.DataFrame({'a': [1, 2, 3], 'b': [4, 5, 6]})
+
+wf.init_state({
```
Comment (FabienArcellier) [inline docs/framework/dataframe.mdx:78]: move at the top

File: docs/framework/dataframe.mdx [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+#### Handle events from a dataframe editor
+
+**The dataframe editor emits events when an action is performed**. You must subscribe to events to integrate changes to the state of the application.
+
+```python
+import pandas
+import writer as wf
+
+# Subscribe this event handler to the `wf-dfeditor-add` event
+def on_record_add(state, payload):
```
Comment (FabienArcellier) [inline docs/framework/dataframe.mdx:54]: document the method to get record and to manage selection

File: src/writer/core.py (hunk 1 above)
Comment (FabienArcellier) [reply src/writer/core.py:1636]: I wrote an `import_failure` decorator to handle this type of requirement.

```python
    @staticmethod
    @import_failure(rvalue=False)
    def match(df: Any) -> bool:
        import pandas
        return True if isinstance(df, pandas.DataFrame) else False
```

File: docs/framework/dataframe.mdx [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+**writer framework places the dataframe at the core of the application**. This is a great way for modeling a complex and massive data system.
+it offers components as `dataframe` to manipulate dataframes. These components allow you to visualize and interact with dataframes.
+
+| compatibility      | dataframe                             |
+|--------------------|---------------------------------------|
+| `pandas.DataFrame` | x                                     |
+| `polar.DataFrame`  | x                                     |
+| `list of records`  | x (with `EditableDataframe`)          |
+
+### Use a dataframe
```
Comment (madeindjs) [inline docs/framework/dataframe.mdx:14]: The base headline level should be 2 as I see for other documents

https://github.com/writer/writer-framework/blob/97a5948a8afb68e5c10ffd4797cbc049f55125bb/docs/framework/ai-module.mdx?plain=1#L7

**Pull request merged**